features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/DeviceConfigServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.deviceconfig.service;
+
+import joptsimple.internal.Strings;
+import org.opennms.features.deviceconfig.persistence.api.ConfigType;
+import org.opennms.netmgt.config.PollerConfig;
+import org.opennms.netmgt.config.ReadOnlyPollerConfigManager;
+import org.opennms.netmgt.config.poller.Package;
+import org.opennms.netmgt.dao.api.IpInterfaceDao;
+import org.opennms.netmgt.dao.api.SessionUtils;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.poller.LocationAwarePollerClient;
+import org.opennms.netmgt.poller.MonitoredService;
+import org.opennms.netmgt.poller.PollerResponse;
+import org.opennms.netmgt.poller.ServiceMonitor;
+import org.opennms.netmgt.poller.ServiceMonitorAdaptor;
+import org.opennms.netmgt.poller.support.SimpleMonitoredService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class DeviceConfigServiceImpl implements DeviceConfigService {
+
+    private static final String DEVICE_CONFIG_PACKAGE_NAME = "device-config";
+    private static final Logger LOG = LoggerFactory.getLogger(DeviceConfigServiceImpl.class);
+    private static final Map<ConfigType, String> deviceConfigServiceMap = new HashMap<>();
+
+    static {
+        deviceConfigServiceMap.put(ConfigType.Default, "DeviceConfig-Default");
+        deviceConfigServiceMap.put(ConfigType.Running, "DeviceConfig-Running");
+    }
+
+    @Autowired
+    private LocationAwarePollerClient locationAwarePollerClient;
+
+    @Autowired
+    private SessionUtils sessionUtils;
+
+    @Autowired
+    private IpInterfaceDao ipInterfaceDao;
+
+    @Qualifier("deviceConfigMonitorAdaptor")
+    private ServiceMonitorAdaptor serviceMonitorAdaptor;
+
+
+    @Override
+    public void triggerConfigBackup(String ipAddress, String location, String configType) throws IOException {
+
+        if (!deviceConfigServiceMap.containsKey(ConfigType.valueOf(configType))) {
+            throw new IllegalArgumentException("Unknown configType " + configType);
+        }
+        String serviceName = deviceConfigServiceMap.get(ConfigType.valueOf(configType));
```

Comment:
We do not need that logic if we agree that the service name = `DeviceConfig-<configType>`.

Reply:
I feel It's better to have a static map than deriving in the code.

Reply:
If we don't want to rely on a convention that maps service names into config types then we should just iterate over the services contained in the `device-config` package and register a mapping from configured config types into service names.

Reply:
For manual trigger, all the service parameters need to be fetched from service metadata [here](https://github.com/OpenNMS/opennms/blob/0fcbc5a092113a9f67103f0efcdb217d9142d7de/features/poller/client-rpc/src/main/java/org/opennms/netmgt/poller/client/rpc/PollerRequestBuilderImpl.java#L140).

Why would there be multiple services for the same `configType` on a given interface ?